### ISSUE TITLE
The backdrop animation doesn't work, background  appears only after reaching top position of the bottom sheet modal

### ISSUE THREAD

**Issue by andreibahachenka** [gorhom/react-native-scroll-bottom-sheet#846]
# Bug

Hey all!
The backdrop animation doesn't work, background appears only after reaching top position of the bottom sheet modal , it happened after updating bottom-sheet to 4.1.5 version

My custom backdrop looks absolutely the same as in the example

## Environment info

| Library                         | Version |
| ------------------------------- | ------- |
| @gorhom/bottom-sheet            | ^4.1.5   |
| react-native                    | 0.64.3   |
| react-native-reanimated         | ^2.3.1   |
| react-native-gesture-handler    | ~2.1.0   |

## Steps To Reproduce

1. Open backdropsheetmodal 

Describe what you expected to happen:

1. Animation smoothly changes the opacity, but background of the bottomSheetBackdrop appears immediately after opening bottomSheetModal

## Reproducible sample code

i guess that the problem can be somewhere here:
```
import React, { useMemo } from 'react'
import { BottomSheetBackdropProps } from '@gorhom/bottom-sheet'
import Animated, { Extrapolate, interpolate, useAnimatedStyle } from 'react-native-reanimated'

import { theme } from '../constants/'

// https://gorhom.github.io/react-native-bottom-sheet/custom-backdrop
const CustomBackdrop = ({ animatedIndex, style }: BottomSheetBackdropProps) => {
    const containerAnimatedStyle = useAnimatedStyle(() => ({
        opacity: interpolate(animatedIndex.value, [0, 0.6], [0, 0.6], Extrapolate.CLAMP),
    }))

    const containerStyle = useMemo(
        () => [
            style,
            {
                backgroundColor: theme.colors.shadow,
            },
            containerAnimatedStyle,
        ],
        [style, containerAnimatedStyle],
    )

    return <Animated.View style={containerStyle} />
}

export default CustomBackdrop

```

**Comment by andreibahachenka** [comment]
@gorhom Hey Gorhom! maybe you are able to take a look?

**Comment by andreibahachenka** [comment]
@gorhom how we can just close the modal with animation? 
when i don't provide the first value in snappoints, it gives me an error: index should be -1 or 0 (no animation)
if i provide the first value in snap points as 0, it shows me another error: you have to use enablePanDownToClose={true}, but i already use it

all i need is to close bottomsheetmodal with animation of backdrop component


```
<BottomSheetModal
                            ref={bottomSheetRef}
                            index={1}
                            snapPoints={['65%']}
                            enablePanDownToClose={true}
                            style={styles.bottomSheet}
                            backdropComponent={BottomSheetCustomBackdrop}
                        >
```

**Comment by andreibahachenka** [comment]
found workaround to check on handleSheetChanges and dismiss the modal

**Comment by gudberg** [comment]
Can you be more specific about the solution @b0n1q? Also happening in my application.

**Comment by white9ja** [comment]
remove the Animated from react-native and import it from 
react-native-animated.
for instance
import { View, Text, TouchableOpacity, Image, Platform, Animated} from "react-native";  < --(remove the Animated here)

 import Animated,  {
    Easing, Extrapolate,
    interpolate,
    useAnimatedStyle,
} from 'react-native-reanimated';  < --- (Add the Animated here)

**Comment by JamieGoodson** [comment]
For anyone stumbling into this in 2025 or beyond, here's how I fixed the issue:

1. Use example `CustomBackdrop` component as provided in the official docs here: https://gorhom.dev/react-native-bottom-sheet/custom-backdrop
2. Ensure your imports are **exactly as provided** in the above doc. Your IDE will try to import some from `react-native`, but they must all come from `react-native-reanimated` (as @white9ja suggested above)

However I can only seem to get this to work if the modal has multiple snap points. With just one snap point (ie the modal is always full height), the backdrop doesn't show/animate. Hoping anyone has any ideas?

Edit: Found a workaround by providing two snap points and abusing `onAnimate` to force the sheet to always snap to index `1`.

```
    <BottomSheetModal
      ref={modalRef}
      snapPoints={[100, 500]}
      backgroundStyle={styles.sheet}
      backdropComponent={CustomBackdrop}
      onAnimate={(index) => {
        if (index !== 1) {
          modalRef.current?.snapToIndex(1); // Will always snap to 500
        }
      }}
    >
```

Have created a separate ticket for this here: https://github.com/gorhom/react-native-bottom-sheet/issues/2281

**Comment by ezescigo** [comment]
> For anyone stumbling into this in 2025 or beyond, here's how I fixed the issue:
> 
> 1. Use example `CustomBackdrop` component as provided in the official docs here: https://gorhom.dev/react-native-bottom-sheet/custom-backdrop
> 2. Ensure your imports are **exactly as provided** in the above doc. Your IDE will try to import some from `react-native`, but they must all come from `react-native-reanimated` (as [@white9ja](https://github.com/white9ja) suggested above)
> 
> However I can only seem to get this to work if the modal has multiple snap points. With just one snap point (ie the modal is always full height), the backdrop doesn't show/animate. Hoping anyone has any ideas?
> 
> Edit: Found a workaround by providing two snap points and abusing `onAnimate` to force the sheet to always snap to index `1`.
> 
> ```
>     <BottomSheetModal
>       ref={modalRef}
>       snapPoints={[100, 500]}
>       backgroundStyle={styles.sheet}
>       backdropComponent={CustomBackdrop}
>       onAnimate={(index) => {
>         if (index !== 1) {
>           modalRef.current?.snapToIndex(1); // Will always snap to 500
>         }
>       }}
>     >
> ```
> 
> Have created a separate ticket for this here: [#2281](https://github.com/gorhom/react-native-bottom-sheet/issues/2281)

Thanks for the workaround! In that case we cannot use the enableDynamicSizing either right?
